lib/Gateway.js

```diff
@@ -588,15 +588,21 @@ Gateway.prototype.discoverDevice = function (node, hassDevice) {
 
         var mode = node.values[payload.mode_state_topic]
         var currTemp = node.values[payload.current_temperature_topic]
-        var setId = hassDevice.setpoint_topic && hassDevice.setpoint_topic[mode.value] ? hassDevice.setpoint_topic[mode.value] : hassDevice.default_setpoint
+        var setId;
+        if(mode !== undefined) {
+          setId = hassDevice.setpoint_topic && hassDevice.setpoint_topic[mode.value] ? hassDevice.setpoint_topic[mode.value] : hassDevice.default_setpoint
+        }
+        else {
+          setId = hassDevice.default_setpoint;
+        }
         var setpoint = node.values[setId]
-
         var map = getMappedValues(mode, hassDevice.mode_map, this.config.integerList)
```

Comment:
Why don't move this end also all the content in the next `if` inside the first `if(mode !== undefined)` ? The code would be much cleaner

Reply:
Indeed, the code would be much cleaner if the check is only performed once. (Not sure why I thought two if  statements would be necessary).
Updated MR accordingly.